tests/unit/s2n_rsa_pss_rsae_test.c

```diff
@@ -245,11 +247,17 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_asn1der_to_public_key_and_type(&rsa_pss_public_key, &rsa_pss_pkey_type, &rsa_pss_cert_chain->cert_chain->head->raw));
             EXPECT_EQUAL(rsa_pss_pkey_type, S2N_PKEY_TYPE_RSA_PSS);
 
-            /* Set the keys equal */
+            /* In openssl 3 the function `EVP_PKEY_get1_RSA` returns a cached copy of PKEY
```

Comment:
This comment won't really make sense after this PR is merged. Can you write the comment as though you're explaining what the code is doing and not the difference between Openssl 3.0.0 and other versions?

Reply:
https://github.com/aws/s2n-tls/pull/3473/commits/e3fa79b5cc3cf581cc4140baa47d28d19ca5983c